jetty-util/src/test/java/org/eclipse/jetty/util/HostPortTest.java

```diff
@@ -26,62 +26,122 @@
 
 import static org.hamcrest.MatcherAssert.assertThat;
 import static org.hamcrest.Matchers.is;
-import static org.junit.jupiter.api.Assertions.assertNull;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
 public class HostPortTest
 {
     private static Stream<Arguments> validAuthorityProvider()
     {
         return Stream.of(
-            Arguments.of("", "", null),
-            Arguments.of(":80", "", "80"),
+            // Common normal Host headers
             Arguments.of("host", "host", null),
+            Arguments.of("[0::0::0::0::1]:", "[0::0::0::0::1]", null),
             Arguments.of("host:80", "host", "80"),
             Arguments.of("10.10.10.1", "10.10.10.1", null),
             Arguments.of("10.10.10.1:80", "10.10.10.1", "80"),
             Arguments.of("[0::0::0::1]", "[0::0::0::1]", null),
             Arguments.of("[0::0::0::1]:80", "[0::0::0::1]", "80"),
             Arguments.of("0:1:2:3:4:5:6", "[0:1:2:3:4:5:6]", null),
             Arguments.of("127.0.0.1:65535", "127.0.0.1", "65535"),
+            // Whitespace
+            Arguments.of("host   ", "host", null),
+            Arguments.of("    host   ", "host", null),
+            Arguments.of("host   :777", "host", 777),
+            Arguments.of("    host   :777", "host", 777),
+            Arguments.of("    host   :   777", "host", 777),
+            Arguments.of("    host   :777   ", "host", 777),
+            Arguments.of("    host   :   777   ", "host", 777),
```

Comment:
These whitespace tests are odd, on one hand it's part of the "be lenient in what you accept", on the other hand, it's not to spec.

WDYT?

Reply:
were we forgiving before?  If not, then lets not start being forgiving. If we were, then let's keep doing so

Reply:
we were not forgiving before.
I'll move these to invalid.

Reply:
Here's how to describe the old behavior...

* If the whitespace is in the _host_ portion of `HostPort`, it is preserved, and does not result in a 400 Bad Request, the request will reach the application, and use of various APIs will result in returning this whitespace, even secondary APIs will fail due to an invalid URL/URI during `getRequestURL()` or `sendRedirect()`.
* If the whitespace is in the _port_ portion of `HostPort`, the internal Integer.parseInt() fails with a `java.lang.NumberFormatException: For input string: "   777   "` and causes a 400 Bad Request.